hydra/core/override_parser/overrides_parser.py

```diff
@@ -28,6 +28,21 @@
     )
     sys.exit(1)
 
+# Mappings between lexer modes and parser rules.
+MODE_TO_RULES = {
+    "KEY": ["key", "override", "package", "packageOrGroup"],
+    "VALUE": [
```

Comment:
would be simpler to go with KEY for the small subset of rules that are in the KEY mode and default to VALUE for the rest.

Reply:
Simplified in 13269391